src/Nethermind/Nethermind.Blockchain/Producers/BlockProducerBase.cs

```diff
@@ -187,8 +194,24 @@ protected virtual Block PrepareBlock(BlockHeader parent)
             return block;
         }
 
-        protected virtual long GetGasLimit(BlockHeader parent) => parent.GasLimit;
+        protected virtual long GetGasLimit(BlockHeader parent)
+        {
+            long parentGasLimit = parent.GasLimit;
+            long gasLimit = parentGasLimit;
+            
+            long? targetGasLimit = _miningConfig.TargetBlockGasLimit;
+            if (targetGasLimit != null)
+            {
+                IReleaseSpec spec = _specProvider.GetSpec(parent.Number + 1);
+                long maxGasLimitDifference = parentGasLimit / spec.GasLimitBoundDivisor;
+                gasLimit = targetGasLimit.Value > parentGasLimit
+                    ? parentGasLimit + Math.Min(targetGasLimit.Value - parentGasLimit, maxGasLimitDifference)
+                    : parentGasLimit - Math.Min(parentGasLimit - targetGasLimit.Value, maxGasLimitDifference);
+            }
+
+            return gasLimit;
+        }
```

Comment:
I would put it behind some IGasLimitCalculator, make GasLimitOverride in AuRa just a type of decorator over this calculator

Reply:
not touching the AuRa gas limit override, not needed here, also mine approach is simpler than introducing other classes